application-analytics-ui/src/main/resources/Analytics/Code/Macros/RowEvolution.xml

```diff
@@ -0,0 +1,289 @@
+    const field = data.map(item =&gt; item[$('#field').val()] || 0);
+    const ctx = $('#chart-display');
+    if (chart) {
+      chart.destroy();
+    }
+    chart = new Chart(ctx, {
+      type: 'line',
+      data: {
+        labels: dates,
+        datasets: [{
+          label: '',
+          data: field,
+          borderColor: 'rgba(75, 192, 192, 1)',
+          backgroundColor: 'rgba(75, 192, 192, 0.2)',
+          fill: false
+        }]
+      },
+      options: {
+        scales: {
+          y: {
+            beginAtZero: true
+          }
+        }
+      }
+    });
+    $('#chartModal').modal('show');
+  };
+
+  /**
+   * Fetches the data from Matomo using the XWiki server as a proxy through the RowEvolutionJSON page. It then saves the data in session 
+   * storage for easy access. This allows the user to display a different field for the same time interval without making a new request and updates the chart.
+   *
+   * @param parameter parameter need for the request
+   */
+  const fetchChartData = (parameter) =&gt; {
+    $('#options').data('parameter', parameter);
+    const options = $('#options')
+    const period = options.data('period')
+    const date = options.data('date');
+    const requestPage = options.data('requestPage')
+    $.getJSON(`${requestPage}?parameter=${parameter}&amp;period=${period}&amp;date=last${date}`)
+      .done((data) =&gt; {
+        window.sessionStorage.setItem('analyticsLastRequestCache', JSON.stringify(data));
+        renderChart(data);
+      })
+      .fail((jqxhr, textStatus, error) =&gt; {
+        const err = `${textStatus}, ${error}`;
+        console.error(`Request Failed: ${err}`);
+      });
+  };
+
+  /**
+   * Will show only the time options available for the period given as a parameter and updates the options
+   * 
+   * @param period Day/Week/Month/Year
+   */
+  const displayPeriodDurations = (period) =&gt; {
+    PERIODS.forEach(p =&gt; {
+      p === period ? $(`#periodDuration${p}`).show() : $(`#periodDuration${p}`).hide();
+    });
+    $('#options').data('period', period.toLowerCase());
+    $('#options').data('date', $(`#periodDuration${period}`).val());
+  };
+
+  /**
+   * Resets to the default option for all the selectos when the modal is closed.
+   */
+  const restOptions = () =&gt; {
+    $('#chartModal').on('hidden.bs.modal', function () {
+      PERIODS.forEach(p =&gt; {
+        $(`#periodDuration${p}`).prop('selectedIndex', 0);
+      });
+      $('#field').prop('selectedIndex', 0);
+    });
+  };
+
+  const displayStatisitcs = () =&gt; {
+    $('#field').on('change', function () {
+      renderChart(JSON.parse(window.sessionStorage.getItem('analyticsLastRequestCache')));
+    });
+  };
+
+  /**
+   * Binds the event that will open the modal and display the data when the 'Row Evolution' button is clicked. 
+   */
+  const bindRowEvolutionDisplay = () =&gt; {
+    $(document).on('click', '.displayMostViewedPagesActions', function () {
```

Comment:
You don't need to have this event in a separate method that is called only on document ready. You can simply add it as you added the `$(document).ready(...` and it will be registered.
Maybe you would have needed the extra steps in case the event was `$('.myClass').on('click...`, because here `$('.myClass')` is the selector and this needs to be ready when you bind the event. But in your case `$(document)` is already loaded, and `'.displayMostViewedPagesActions'` will be loaded as well when the click is done

Also, you need a better selector for you button. I suggest you also use the livetable class (and add it in case it does not exist) besides the button one, to be sure it's inside the livetable